datafusion/physical-expr/src/expressions/any.rs

```diff
@@ -0,0 +1,674 @@
+        list: Arc<dyn PhysicalExpr>,
+    ) -> Self {
+        Self { value, op, list }
+    }
+
+    /// Compare for specific utf8 types
+    fn compare_utf8<T: OffsetSizeTrait>(
+        &self,
+        array: ArrayRef,
+        list: &ListArray,
+        negated: bool,
+        list_from_scalar: bool,
+    ) -> Result<ColumnarValue> {
+        let array = array
+            .as_any()
+            .downcast_ref::<GenericStringArray<T>>()
+            .unwrap();
+
+        if negated {
+            Ok(ColumnarValue::Array(Arc::new(neq_utf8(
+                array,
+                list,
+                list_from_scalar,
+            )?)))
+        } else {
+            Ok(ColumnarValue::Array(Arc::new(eq_utf8(
+                array,
+                list,
+                list_from_scalar,
+            )?)))
+        }
+    }
+
+    /// Get the left side of the binary expression
+    pub fn left(&self) -> &Arc<dyn PhysicalExpr> {
+        &self.value
+    }
+
+    /// Get the right side of the binary expression
+    pub fn right(&self) -> &Arc<dyn PhysicalExpr> {
+        &self.list
+    }
+
+    /// Get the operator for this binary expression
+    pub fn op(&self) -> &Operator {
+        &self.op
+    }
+
+    /// Compare for specific utf8 types
+    fn compare_bool(
+        &self,
+        array: ArrayRef,
+        list: &ListArray,
+        negated: bool,
+        list_from_scalar: bool,
+    ) -> Result<ColumnarValue> {
+        let array = array.as_any().downcast_ref::<BooleanArray>().unwrap();
+
+        if negated {
+            Ok(ColumnarValue::Array(Arc::new(neq_bool(
+                array,
+                list,
+                list_from_scalar,
+            )?)))
+        } else {
+            Ok(ColumnarValue::Array(Arc::new(eq_bool(
+                array,
+                list,
+                list_from_scalar,
+            )?)))
+        }
+    }
+}
+
+impl std::fmt::Display for AnyExpr {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{} {} ANY({})", self.value, self.op, self.list)
+    }
+}
+
+impl PhysicalExpr for AnyExpr {
+    /// Return a reference to Any that can be used for downcasting
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn data_type(&self, _input_schema: &Schema) -> Result<DataType> {
+        Ok(DataType::Boolean)
+    }
+
+    fn nullable(&self, input_schema: &Schema) -> Result<bool> {
+        self.value.nullable(input_schema)
+    }
+
+    fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
+        let value = match self.value.evaluate(batch)? {
+            ColumnarValue::Array(array) => array,
+            ColumnarValue::Scalar(scalar) => scalar.to_array(),
+        };
+
+        let (list, list_from_scalar) = match self.list.evaluate(batch)? {
+            ColumnarValue::Array(array) => (array, false),
+            ColumnarValue::Scalar(scalar) => (scalar.to_array(), true),
+        };
+        let as_list = list
```

Comment:
I'm probably missing something fundamental, but I'm confused as to why this is a ListArray, i.e. 2-dimensional.

In all the examples given the expression is `[1,2,3]` which is 1 dimensional?